documentation/src/docs/controller-service.md

```diff
@@ -116,24 +125,43 @@ retention and scaling.
 have two policies that users can define, namely [**Scaling** **Policy**](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/ScalingPolicy.java) and
 [**Retention** **Policy**](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/RetentionPolicy.java).
 
-       - **Scaling policy** describes if and under what circumstances a Stream should automatically scale its number of segments.  
+       - **Scaling policy** describes if and under what circumstances a Stream should automatically scale its number of Segments.  
        - **Retention policy** describes a policy about how much data to retain within a Stream based on **time** (*Time-based Retention*) and data **size** (*Size-based Retention*).
 
   3. [**Transaction**](pravega-concepts.md#transactions) **Management**: Implementing Transactions requires the manipulation of Stream Segments. With
-each Transaction, Pravega creates a set of Transaction segments, which
+each Transaction, Pravega creates a set of Transaction Segments, which
 are later merged onto the Stream Segments upon commit or discarded upon
 aborts. The Controller performs the role of Transaction manager and is
 responsible for creating and committing Transactions on a given Stream.
 Upon creating Transactions, Controller also tracks Transaction timeouts
 and aborts transactions whose timeouts have elapsed. Details of
 Transaction management can be found later in the [Transactions](#transaction-manager) section.
 
+  4. [**Watermarks**](http://pravega.io/docs/v0.8.0/watermarking/): Watermarks are a data structure produced by Controller that refer to a unique position in the Stream and associates time information corresponding to that position. The watermark is computed by Controller by coordinating the time and position information reported by different writer applications and generating a time window. 
```

Comment:
Should this link be with `latest` instead to associated with a specific version? `http://pravega.io/docs/v0.8.0/watermarking/`